src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -601,7 +601,7 @@ public bool IsNamedValueOfKind(string kind)
                     ? 1 +
                       (ChildCount == 0
                           ? 0
-                          : GetNames(DPath.Root)
+                          : GetAllNames(DPath.Root)
```

Comment:
>GetAllNames [](http://example.com/codeflow?start=28&length=11)

This is definitely subtle - is there a test case we can add for it in core (preferably a .txt one if possible)?

Reply:
fwiw, I agree with this fix - we just need a test for it. 

Max() is throwing an exception because GetNames() is returning empty despite ChildCount being > 0. That's happening because GetNames() is GetallNames() with a filter. So remove the filter and just use GetAllNames().